doc/dev/BUILD.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Building TileDB from source
+---
+
+## Prerequisites
+
+* CMake 3.21 or greater
+* C++ compiler with C++20 support
+    * The minimum compiler versions TileDB is being tested in CI are:
+        * MSVC from Visual Studio 2019 16.11
+        * GCC 10.3 on Linux and Windows via MinGW
+        * Apple Clang 14
+* Git
```

Comment:
Git is required by vcpkg, regardless of whether it was downloaded by us or provided by the user.

Reply:
You also need `curl`

Reply:
Updated.